components/Themes/src/MDCTonalPalette.h

```diff
@@ -0,0 +1,73 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+/**
+ A tonal palette consists of a main color and variants of the main color that are lighter and darker
+ shades of the main color. Material design guidelines recommend one main color with nine color
+ variations. One of the color variations is designated as a light color in relation to the main
+ color and another of the color variations is designated as a dark color in relation to the main
+ color.
+ */
+@interface MDCTonalPalette : NSObject
```

Comment:
Need to discuss this with you. NSCopying protocol may require the properties to be mutable.